Composer/packages/client/src/pages/publish/Publish.tsx

```diff
@@ -159,49 +159,85 @@ const Publish: React.FC<RouteComponentProps<{ projectId: string; targetName?: st
   };
 
   const [pendingNotification, setPendingNotification] = useState<Notification>();
+  const [previousBotPublishHistoryList, setPreviousBotPublishHistoryList] = useState(botPublishHistoryList);
   // check history to see if a 202 is found
   useEffect(() => {
     // most recent item is a 202, which means we should poll for updates...
     selectedBots.forEach((bot) => {
-      if (bot.publishTarget && bot.publishTargets) {
-        const selectedTarget = bot.publishTargets.find((target) => target.name === bot.publishTarget);
-        const botProjectId = bot.id;
-        if (selectedTarget) {
-          const botPublishHistory = botPublishHistoryList.find(
-            (publishHistory) => publishHistory.projectId === botProjectId
-          )?.publishHistory[bot.publishTarget];
-          if (botPublishHistory && botPublishHistory.length > 0) {
-            if (botPublishHistory[0].status === 202) {
-              getUpdatedStatus(selectedTarget, botProjectId);
-            } else if (botPublishHistory[0].status === 200 || botPublishHistory[0].status === 500) {
-              bot.status = botPublishHistory[0].status;
-              if (showNotifications[bot.id]) {
-                pendingNotification && deleteNotification(pendingNotification.id);
-                addNotification(createNotification(getPublishedNotificationCardProps(bot)));
-                setShowNotifications({ ...showNotifications, [botProjectId]: false });
-              }
-            } else if (selectedTarget && selectedTarget.lastPublished && botPublishHistory.length === 0) {
-              // if the history is EMPTY, but we think we've done a publish based on lastPublished timestamp,
-              // we still poll for the results IF we see that a publish has happened previously
-              getPublishStatus(botProjectId, selectedTarget);
-            }
-            setBotStatusList(
-              botStatusList.map((item) => {
-                if (item.id === botProjectId) {
-                  item.status = botPublishHistory[0].status;
-                  item.comment = botPublishHistory[0].comment;
-                  item.message = botPublishHistory[0].message;
-                  item.time = botPublishHistory[0].time;
-                }
-                return item;
-              })
-            );
+      if (!(bot.publishTarget && bot.publishTargets)) {
```

Comment:
First, the comments "// most recent item is a 202, which means we should poll for updates..." is not related to this line.  there is even no 202 in many lines. 

For such a long use effect and the many other long use effects below, we should give a top-level summary.